packages/blade/src/components/Typography/BaseText/__tests__/BaseText.native.test.tsx

```diff
@@ -0,0 +1,287 @@
+import React from 'react';
+import renderWithTheme from '../../../../_helpers/testing/renderWithTheme.native';
+import BaseText from '../';
+import { ERROR_AS_PROP_NOT_SUPPORTED } from '../StyledBaseText.native';
+import { getInvalidPropValueError } from '../BaseText';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<BaseText />', () => {
+  it('should render regular text', () => {
+    const displayText = 'Displaying some text';
+    const renderedTree = renderWithTheme(
+      <BaseText
+        color="surface.text.normal.highContrast"
+        fontFamily="text"
+        lineHeight="m"
+        fontSize={25}
+        fontWeight="regular"
+      >
+        {displayText}
+      </BaseText>,
+    ).toJSON();
+    expect(renderedTree).toMatchSnapshot();
+  });
+
+  it('should render text in italics style with line-through', () => {
+    const displayText = 'Displaying some text';
+    const renderedTree = renderWithTheme(
+      <BaseText
+        color="surface.text.normal.highContrast"
+        fontFamily="text"
+        lineHeight="m"
+        fontSize={25}
+        fontWeight="regular"
+        fontStyle="italic"
+        textDecorationLine="line-through"
+      >
+        {displayText}
+      </BaseText>,
+    ).toJSON();
+    expect(renderedTree).toMatchSnapshot();
+  });
+
+  it('should throw error when "as" prop passed', () => {
+    try {
+      const displayText = 'Displaying some text';
+      renderWithTheme(
+        <BaseText
+          color="surface.text.normal.highContrast"
+          fontFamily="text"
+          lineHeight="m"
+          fontSize={25}
+          fontWeight="regular"
+          fontStyle="italic"
+          textDecorationLine="line-through"
+          as="p"
+        >
+          {displayText}
+        </BaseText>,
+      ).toJSON();
```

Comment:
Why do you need to do `toJSON` here?
```suggestion
      );
```